### PR TITLE
fix(pagination): conform next/prev button to ghost, icon size to 16

### LINKED ISSUE #6057: [Pagination]: icon size in buttons should be 16
## What package(s) are you using?

- [ ] `carbon-components`
- [x] `carbon-components-react`

## Detailed description

The pagination component pagination buttons use caret icons of size `24` when typical Buttons w/ `hasIconOnly` use `16`. Is there a reason pagination uses a larger size?

Present in Carbon v10.10.3 (and latest)

For `carbon-addons-iot-react`, this will help get all icons in buttons consistent in size on our table.

### PR DESCRIPTION
**Pull request by tay1orjones**
Closes #6057 

Refactor pagination to use ghost button w/ size 16 icons for next/prev

#### Changelog

**Changed**

- pagination component to use ghost button
- updated styling

#### Testing / Reviewing

I'd definitely appreciate a review of the styling, particularly the focus styling. There's some specificity issues with the default ghost button styles I had to work around. I think overall it's valid, but it feels like the situation with the borders could be streamlined?

### CODE REVIEW

**Comment by joshblack** [comment]
Definitely would be interested to see what the correct spec is here, it seems like from the image  on the style tab the icons should be 20/20: https://www.carbondesignsystem.com/components/pagination/style but will wait for @jeanservaas to confirm 👀

**Review by joshblack — COMMENTED** [review]
Left a couple above 👀

**Comment by tay1orjones** [comment]
No sweat if this PR ends up in the trash. Was an anomaly I noticed as I reviewed the size of icons in the table toolbar (16). Feels like this should be consistent, but as you point out the spec says 20.

**Comment by jeanservaas** [comment]
@tay1orjones @joshblack @asudoh 

Can somebody please post a screen grab here of the status quo with the bug and the desired state? 

In general, I'm not seeing them in the issues or the PRs and this makes things very difficult for visual designers to review.

**Comment by asudoh** [comment]
@tay1orjones Seems that you expect no visual change with your change, is it correct? Otherwise, could you please list what visual change you meant to make? Thanks!

**Comment by tay1orjones** [comment]
@jeanservaas Yep, sorry for the lack of clarity. The size of the caret icon in the next/prev buttons is decreased to 16 to fall in line with styling of ghost icon-only buttons.

Current ([link](https://react.carbondesignsystem.com/?path=/story/pagination--%E2%86%AA%EF%B8%8E-multiple-pagination-components)):
image

Proposed in this PR ([link](https://deploy-preview-6058--carbon-components-react.netlify.app/?path=/story/pagination--%E2%86%AA%EF%B8%8E-multiple-pagination-components)):
image

**Review by jeanservaas — APPROVED** [review]
OK! Thanks @tay1orjones for the quick turnaround on those grabs, you are a champ!!!... sorry... that was grumpy pre-coffee me this morning. 

I think I know what happened here @joshblack 

Chris Connors modified the carrot when Tyler was working on the new tree component. Mike asked him to make the asset larger on the canvas. As a result, they now look massive at 20px. And this is an awesome catch and update. Boom!

File: packages/react/src/components/Pagination/Pagination.js [hunk 1]
```diff
@@ -313,22 +314,28 @@ export default class Pagination extends Component {
               ? pageText(statePage)
               : pageRangeText(statePage, totalPages)}
           </span>
-          <button
-            type="button"
+          <Button
+            kind={'ghost'}
```
Comment (joshblack) [inline packages/react/src/components/Pagination/Pagination.js]:
```suggestion
            kind="ghost"
```

File: packages/react/src/components/Pagination/Pagination.js [hunk 2]
```diff
@@ -313,22 +314,28 @@ export default class Pagination extends Component {
...
             className={backButtonClasses}
+            hasIconOnly
+            renderIcon={CaretLeft16}
+            iconDescription={backwardText}
+            tooltipAlignment={'center'}
```
Comment (joshblack) [inline packages/react/src/components/Pagination/Pagination.js]:
```suggestion
            tooltipAlignment="center"
```

File: packages/react/src/components/Pagination/Pagination.js [hunk 3]
```diff
@@ -313,22 +314,28 @@ export default class Pagination extends Component {
...
+            tooltipPosition={'top'}
```
Comment (joshblack) [inline packages/react/src/components/Pagination/Pagination.js]:
```suggestion
            tooltipPosition="top"
```

File: packages/react/src/components/Pagination/Pagination.js [hunk 4]
```diff
@@ -313,22 +314,28 @@ export default class Pagination extends Component {
...
-            aria-label={backwardText}
-            disabled={backButtonDisabled}>
-            <CaretLeft24 />
-          </button>
-          <button
-            type="button"
+            disabled={backButtonDisabled}
+          />
+          <Button
+            kind={'ghost'}
```
Comment (joshblack) [inline packages/react/src/components/Pagination/Pagination.js]:
```suggestion
            kind="ghost"
```

File: packages/react/src/components/Pagination/Pagination.js [hunk 5]
```diff
@@ -313,22 +314,28 @@ export default class Pagination extends Component {
...
             className={forwardButtonClasses}
-            aria-label={forwardText}
+            hasIconOnly
+            renderIcon={CaretRight16}
+            iconDescription={forwardText}
+            tooltipAlignment={'end'}
```
Comment (joshblack) [inline packages/react/src/components/Pagination/Pagination.js]:
```suggestion
            tooltipAlignment="end"
```

File: packages/react/src/components/Pagination/Pagination.js [hunk 6]
```diff
@@ -313,22 +314,28 @@ export default class Pagination extends Component {
...
+            tooltipPosition={'top'}
```
Comment (joshblack) [inline packages/react/src/components/Pagination/Pagination.js]:
```suggestion
            tooltipPosition="top"
```

**Review by joshblack — CHANGES_REQUESTED** [review]
Couple of style-nits 👀

**Review by dakahn — APPROVED** [review]
Seems cool to me as soon as Josh's changes are sorted 👍

**Review by joshblack — APPROVED** [review]
Seems great! Just have a couple test failures that need to be cleaned up, I think 👀

**Comment by tay1orjones** [comment]
Sorry for the delay here - tests should be passing now 👍

**Comment by tw15egan** [comment]
@tay1orjones Thanks for updating those tests!

**Pull request merged**